haystack/nodes/prompt/invocation_layer/hugging_face_inference.py

```diff
@@ -102,49 +123,78 @@ def invoke(self, *args, **kwargs):
         kwargs_with_defaults.update(kwargs)
         # see https://huggingface.co/docs/api-inference/detailed_parameters#text-generation-task
-        accepted_params = [
-            "top_p",
-            "top_k",
-            "temperature",
-            "repetition_penalty",
-            "max_new_tokens",
-            "max_time",
-            "return_full_text",
-            "num_return_sequences",
-            "do_sample",
-        ]
-        params = {key: kwargs_with_defaults.get(key) for key in accepted_params if key in kwargs_with_defaults}
-        generated_texts = self._post(data={"inputs": prompt, "parameters": params}, **kwargs)
-        if stop_words:
-            for idx, _ in enumerate(generated_texts):
-                earliest_stop_word_idx = len(generated_texts[idx])
-                for stop_word in stop_words:
-                    stop_word_idx = generated_texts[idx].find(stop_word)
-                    if stop_word_idx != -1:
-                        earliest_stop_word_idx = min(earliest_stop_word_idx, stop_word_idx)
-                generated_texts[idx] = generated_texts[idx][:earliest_stop_word_idx]
-
+        params = {
+            "best_of": kwargs_with_defaults.get("best_of", None),
+            "details": kwargs_with_defaults.get("details", True),
+            "do_sample": kwargs_with_defaults.get("do_sample", False),
+            "max_new_tokens": kwargs_with_defaults.get("max_new_tokens", self.max_length),
+            "max_time": kwargs_with_defaults.get("max_time", None),
+            "num_return_sequences": kwargs_with_defaults.get("num_return_sequences", None),
+            "repetition_penalty": kwargs_with_defaults.get("repetition_penalty", None),
+            "return_full_text": kwargs_with_defaults.get("return_full_text", False),
+            "seed": kwargs_with_defaults.get("seed", None),
+            "stop": kwargs_with_defaults.get("stop", stop_words),
+            "temperature": kwargs_with_defaults.get("temperature", None),
+            "top_k": kwargs_with_defaults.get("top_k", None),
+            "top_p": kwargs_with_defaults.get("top_p", None),
+            "truncate": kwargs_with_defaults.get("truncate", None),
+            "typical_p": kwargs_with_defaults.get("typical_p", None),
+            "watermark": kwargs_with_defaults.get("watermark", False),
+        }
+        response = self._post(data={"inputs": prompt, "parameters": params, "stream": stream}, stream=stream)
+        if not stream:
+            output = json.loads(response.text)
+            generated_texts = [o["generated_text"] for o in output if "generated_text" in o]
+        else:
+            handler: TokenStreamingHandler = kwargs_with_defaults.pop("stream_handler", DefaultTokenStreamingHandler())
+            generated_texts = self._process_streaming_response(
+                response=response, stream_handler=handler, stop_words=stop_words
+            )
         return generated_texts
```

Comment:
It's good practice to have the first branch conditional branch of an `if` to run when its expression is `True` rather than false. That lowers the burden the things you need to remember when trying to understand the code.

Also since we're at it we can remove the `else` branch to simplify.

```suggestion
        if stream:
            handler: TokenStreamingHandler = kwargs_with_defaults.pop("stream_handler", DefaultTokenStreamingHandler())
            return self._process_streaming_response(
                response=response, stream_handler=handler, stop_words=stop_words
            )
            
       output = json.loads(response.text)
       return [o["generated_text"] for o in output if "generated_text" in o]
```